src/mongo_odm/query_builder.hpp

```diff
@@ -68,7 +68,7 @@ class ComparisonExpr {
     operator bsoncxx::document::view_or_value() const {
```

Comment:
ah, I meant changing the signature to return a bsoncxx::document::value.

Generally we use that type as a way to bind views or values as arguments to a function (we generous in what we accept).  In terms of what we output, we should be strict (we actually always return a value, make that clear from the function signatures).

I.e.

operator bsoncxx::document::value() const {...}